src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectMembersService.java

```diff
@@ -89,26 +89,53 @@ public String removeUserFromProject(Long projectId, Long userId, Locale locale)
 	/**
 	 * Update a users role on a project
 	 *
-	 * @param projectId - identifier for the current project
-	 * @param userId    - identifier for the user to remove from the project
-	 * @param role      - to update the user to
+	 * @param projectId    - identifier for the current project
+	 * @param userId       - identifier for the user to remove from the project
+	 * @param role         - {@link ProjectRole}  to update the user to
 	 * @param metadataRole - {@link ProjectMetadataRole} to update the user to
-	 * @param locale    - of the currently logged in user
+	 * @param locale       - of the currently logged in user
 	 * @return message to display to the user about the outcome of the change in role.
 	 * @throws UIProjectWithoutOwnerException if removing the user will leave the project without a manager
 	 */
-	public String updateUserRoleOnProject(Long projectId, Long userId, String role, String metadataRole, Locale locale) throws UIProjectWithoutOwnerException {
+	public String updateUserRoleOnProject(Long projectId, Long userId, String role, String metadataRole, Locale locale)
+			throws UIProjectWithoutOwnerException {
 		Project project = projectService.read(projectId);
 		User user = userService.read(userId);
-		ProjectRole projectRole = ProjectRole.fromString(role);
-		ProjectMetadataRole projectMetadataRole = ProjectMetadataRole.fromString(metadataRole);
-		String roleString = messageSource.getMessage("projectRole." + role, new Object[] {}, locale);
+		ProjectRole projectRole = null;
+		ProjectMetadataRole projectMetadataRole;
+		String roleString;
+		String updateSuccessMessage;
+
+		if (!role.isEmpty()) {
+			projectRole = ProjectRole.fromString(role);
+
+			/*
+			 If a user's project role is set to collaborator we drop the metadata restriction to the lowest
+			 level and have the project owner set it accordingly. If a role of owner is set then that user is
+			 given full metadata permissions
+			 */
+			if(projectRole.equals(ProjectRole.PROJECT_USER)) {
```

Comment:
Can we swap this statement around (i.e. check for PROJECT_OWNER role and default that to 4). That way if we add more project roles we don't have to refactor this.

Reply:
Sure..updated in 71aa4b0d602012f4e2131fcbf5257dfd17df64de